thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AuthorizationManager.java

```diff
@@ -129,15 +144,47 @@ public boolean hasRootAccess(final ThirdEyePrincipal principal) {
         accessControl.hasAccess(principal.getAuthToken(), ROOT_RESOURCE_ID, AccessType.WRITE);
   }
 
-  static public <T extends AbstractDTO> ResourceIdentifier resourceId(final T dto) {
+  public <T extends AbstractDTO> ResourceIdentifier resourceId(final T dto) {
+    final AbstractDTO authParentDto = resolveAuthParentDto(dto);
+
     return ResourceIdentifier.from(
-        optional(dto.getId()).map(Objects::toString).orElse(DEFAULT_NAME),
-        optional(dto.getAuth())
+        optional(authParentDto.getId()).map(Objects::toString).orElse(DEFAULT_NAME),
+        optional(authParentDto.getAuth())
             .map(AuthorizationConfigurationDTO::getNamespace).orElse(DEFAULT_NAMESPACE),
-        optional(SubEntities.BEAN_TYPE_MAP.get(dto.getClass()))
+        optional(SubEntities.BEAN_TYPE_MAP.get(authParentDto.getClass()))
             .map(Objects::toString).orElse(DEFAULT_ENTITY_TYPE));
   }
 
+  // Resolves the Authorization parent for certain dto classes.
+  // AnomalyDTO -> EnumerationItemDTO if exists, else AlertDTO
+  // RcaInvestigationDTO -> AnomalyDTO's parent (recursive call)
+  // Everything else -> itself
+  private <T extends AbstractDTO> AbstractDTO resolveAuthParentDto(final T dto) {
```

Comment:
why not just use method overloads here? Doing generics first and then if else defeats the purpose.

```
  private  AbstractDTO resolveAuthParentDto(final AnomalyDTO dto);
  private  AbstractDTO resolveAuthParentDto(final RcaInvestigationDTO dto);
...
```

Reply:
I added the method overloads. I think I still need the if-else in the generic one, since the caller functions don't have the type info and it needs the type-casting.

Reply:
Yeah. then you add this. 
```
  private  AbstractDTO resolveAuthParentDto(final AnomalyDTO dto);
  private  AbstractDTO resolveAuthParentDto(final RcaInvestigationDTO dto);
// then you also have this
  private  AbstractDTO resolveAuthParentDto(final AbstractDTO dto);
```

Java should be able to resolve this. or am I missing something here? 🤔